do_cmake.sh

```diff
@@ -64,11 +64,15 @@ erasure code dir = lib
 EOF
 
 echo done.
-cat <<EOF
+
+if [[ ! $ARGS =~ "-DCMAKE_BUILD_TYPE=RelWithDebInfo" ]] > /dev/null 2>&1 ; then
```

Comment:
i don't think this is correct.

before this change, the warning message is always printed, even if "CMAKE_BUILD_TYPE" is not Debug.

after this change, the warning message is printed, as long as `$ARGS` does not contain "-DCMAKE_BUILD_TYPE=RelWithDebInfo". what if `CMAKE_BUILD_TYPE` is "Release" or "MinSizeRel" ?

i think a better fix is something like
```sh
if [[ ! $ARGS =~ "-DCMAKE_BUILD_TYPE" ]]; then
  cat <<EOF
...
EOF
fi
```

so, this warning message is only printed if user does not specify "CMAKE_BUILD_TYPE".

Reply:
@tchaikov Thanks for your review, and that's a right suggestion, 

Doesn't it matter if the message of the command is displayed as below?
```
+ [[ ! -DWITH_LTTNG=on -DWITH_FIO=on -DWITH_OSD_INSTRUMENT_FUNCTIONS=ON -DWITH_CCACHE=ON =~ -DCMAKE_BUILD_TYPE ]]
```

Reply:
i don't follow you.

> ```
> + [[ ! -DWITH_LTTNG=on -DWITH_FIO=on -DWITH_OSD_INSTRUMENT_FUNCTIONS=ON -DWITH_CCACHE=ON =~ -DCMAKE_BUILD_TYPE ]]
> ```

the bash script above has syntax error. and i cannot tell its intention.

Reply:
@tchaikov 
That was about print messages after executing `do_cmake.sh` script :)
```
-- Found nl: /usr/lib/x86_64-linux-gnu/libnl-3.so  
-- Checking for module 'libcap-ng'
--   Found libcap-ng, version 0.7.9
-- Setting civetweb to use OPENSSL >= 1.1
-- Found RabbitMQ: /usr/include  
-- exclude following files under src: *.js;*.css;civetweb;erasure-code/jerasure/jerasure;erasure-code/jerasure/gf-complete;rocksdb;googletest;spdk;xxHash;isa-l;lua;zstd;crypto/isa-l/isa-l_crypto;blkin;rapidjson;dmclock;seastar;fmt;c-ares
-- Configuring done
-- Generating done
-- Build files have been written to: /home/sungjunyoung/Sources/common/ceph/build
+ cat
+ echo done.
done.
+ [[ ! -DWITH_LTTNG=on -DWITH_FIO=on -DWITH_OSD_INSTRUMENT_FUNCTIONS=ON -DWITH_CCACHE=ON =~ -DCMAKE_BUILD_TYPE ]]
+ cat

****
WARNING: do_cmake.sh now creates debug builds by default. Performance
may be severely affected. Please use -DCMAKE_BUILD_TYPE=RelWithDebInfo
if a performance sensitive build is required.
****
```

The if state expression exposed if I don't pass the output to the `/dev/null`

Reply:
@sungjunyoung i see now. and also i understand why you wanted to redirect the  because we do echo the commands.

probably you could run 
```sh
set +x
```
right before the last `if` block to disable this behavior?

Reply:
Wow,
Maybe,, After ` ${CMAKE} $ARGS "$@" .. || exit 1` looks better for me, looks more clean.

### to-be
```
...
-- Looking for cbreak in /usr/lib/x86_64-linux-gnu/libncurses.so - found
-- Found Curses: /usr/lib/x86_64-linux-gnu/libncurses.so  
-- Found nl: /usr/lib/x86_64-linux-gnu/libnl-3.so  
-- Checking for module 'libcap-ng'
--   Found libcap-ng, version 0.7.9
-- Setting civetweb to use OPENSSL >= 1.1
-- Found RabbitMQ: /usr/include  
-- exclude following files under src: *.js;*.css;civetweb;erasure-code/jerasure/jerasure;erasure-code/jerasure/gf-complete;rocksdb;googletest;spdk;xxHash;isa-l;lua;zstd;crypto/isa-l/isa-l_crypto;blkin;rapidjson;dmclock;seastar;fmt;c-ares
-- Configuring done
-- Generating done
-- Build files have been written to: /home/sungjunyoung/Sources/common/ceph/build
+ set +x
done.

****
WARNING: do_cmake.sh now creates debug builds by default. Performance
may be severely affected. Please use -DCMAKE_BUILD_TYPE=RelWithDebInfo
if a performance sensitive build is required.
****
```